### ISSUE TITLE
[Feature request] Add Created and Updated datetime on all page

### ISSUE THREAD

**Issue by letientai299** [mkdocs/mkdocs#1408]
It's would be nice to see how recent the document is, like:

![image](https://user-images.githubusercontent.com/8386780/36351186-18e6bcdc-14e1-11e8-8f59-6d482c0818ca.png)

The Created and Update datetime information can be get from git repository, based on the first and last commit affect to the file.

Reference: 
- Screenshot is captured from https://cloud.google.com/docs/compare/aws/
- Related issue: https://github.com/squidfunk/mkdocs-material/issues/537

**Comment by waylan** [comment]
Using the git commit date/time is reliant upon using git. Of course, not all MkDocs users use git for their documentation. A specific feature like that would be a good candidate for a third party [Plugin][0].

A more general approach would be to use the "build date" which we already make available via the [`build_date_utc`][2] template variable. The existing themes simply don't display it. For example, the `mkdocs` theme [inserts][1] the date into an HTML comment on the homepage only.  With a little [theme customization][3], one could include that date elsewhere on pages.

In short, if you want the "build date" we don't need to do anything. If you want something more specific, you are welcome to create a plugin to address your specific needs. Again there is no need for MkDocs to do anything.

[0]: http://www.mkdocs.org/user-guide/plugins/#developing-plugins
[1]: https://github.com/mkdocs/mkdocs/blob/master/mkdocs/themes/mkdocs/base.html#L95
[2]: http://www.mkdocs.org/user-guide/custom-themes/#build_date_utc
[3]: http://www.mkdocs.org/user-guide/styling-your-docs/#using-the-theme-custom_dir

**Comment by biochimia** [comment]
Perhaps it would be interesting to have `mkdocs` define a baseline for what version control metadata could look like an how it integrates in the base template. Then third-party plugins could fill in this data knowing it would integrate well with the system.

**Comment by waylan** [comment]
@biochimia actually I think plugin creators should experiment and find what works best. If, at some point in the future, a single approach becomes wildly popular, then we can explore bringing that plugin in-house at that time.

**Comment by GaryPWhite** [comment]
Hey everyone! Looking to use a feature like this, is there any consensus on if this is coming down the pipe?

**Comment by Ir1d** [comment]
@GaryPWhite It seems that mkdocs is not adding this feature, you can call github api by your self actually. My code: https://github.com/24OI/OI-wiki/blob/master/static/extra.js#L33

**Comment by zhaoterryy** [comment]
@GaryPWhite You can use my simple plugin here: [mkdocs-git-revision-date-plugin](https://github.com/zhaoterryy/mkdocs-git-revision-date-plugin)

This will fill in `page.meta.revision_date` allowing you to use it in your templates. It pulls this info from the last time the file was modified and committed to Git.

**Comment by Ir1d** [comment]
@zhaoterryy Hi, how to use your plugin? should I clone the repo to somewhere or install via pip? Thx

**Comment by zhaoterryy** [comment]
Right.. sorry about that. 

`pip install mkdocs-git-revision-date-plugin`

**Comment by zhaoterryy** [comment]
You'll also need to add `git-revision-date` to your list of plugins in `mkdocs.yml`

**Comment by waylan** [comment]
@zhaoterryy you might want to add your plugin to the list on the [wiki][1] so others can find it.

[1]: https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins
